### ISSUE TITLE
YOLOv8n Pose training behavior

### ISSUE THREAD

**Issue by jazzseow** [ultralytics/ultralytics#2112]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I ran `yolo pose train device=0 workers=4 exist_ok=True data=coco-pose.yaml model=yolov8n-pose.yaml epochs=1000 imgsz=640 batch=16 name='pose_coco_v8n_batch16'`

and this is the params
![image](https://user-images.githubusercontent.com/18047843/232990284-e86fc34d-8323-4cf6-9c17-94edfbef9320.png)

is this the right training behavior for COCO?
![image](https://user-images.githubusercontent.com/18047843/232989561-dca3fb31-c426-4ff1-98b0-666674d1b340.png)

### Additional

I also found that the val_batch*_labels.jpg are wrong in the late epoch

val_batch0_labels.jpg, epochs=100
![val_batch0_labels](https://user-images.githubusercontent.com/18047843/232991047-3b3f3d85-6ec0-4682-a92c-ed4f0f76b927.jpg)

val_batch0_labels.jpg, epochs=1
![val_batch0_labels](https://user-images.githubusercontent.com/18047843/232991227-7b7b3838-92ad-4511-99cb-b3dde3630247.jpg)

**Comment by AyushExel** [comment]
@jazzseow hey, the val_ predictions on humans look fine so the training seems to be working. Also, note that coco-pose is actually coco-pose for humans so it should work best only on humans which I think its working.

**Comment by jazzseow** [comment]
@AyushExel , thanks for replying. 

Those are val_batch0_labels.jpg generated after training for 100 epochs

I got this after training from scratch for 100 epoch
![image](https://user-images.githubusercontent.com/18047843/233028434-97640bcc-196e-4a3a-b52e-bd06985f7dc7.png)

and this is the results for the model i downloaded from [yolov8n-pose](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8n-pose.pt)
![image](https://user-images.githubusercontent.com/18047843/233030147-c45c4491-4573-486b-887f-69c8d9dacf41.png)

**Comment by jazzseow** [comment]
I was trying to recreate results from
![image](https://user-images.githubusercontent.com/18047843/233036090-0e6511c9-38a6-448d-b4c5-baf9755db40c.png)
the mAP for bbox is ok but the mAP for pose is far from expected (0.000).

Is it possible to share the training info for your working model?
Yea, meanwhile i will continue to check / play with the hyperparams, and i will update here accordingly

**Comment by glenn-jocher** [comment]
@jazzseow, it's great that you are working towards recreating the results from the image you shared. As you experiment with different hyperparameters, please keep in mind that the performance and accuracy of pose detection is dependent on the quality and diversity of the dataset as well as the model architecture and tuning. It may take some experimentation to find the optimal settings that give you the results you are looking for. Also, it is not possible to share the training information for the working model since we do not have access to the specific information and details that you require without explicitly knowing what model you are referring to.

**Comment by Yangninan** [comment]
val_batch0_labels.jpg, epochs=1
![1681959082131_9938FC84-F182-4945-814E-FA7889B1E3F0](https://user-images.githubusercontent.com/103250675/233246855-7e9f8171-84d9-40b4-a066-4b8a865ff465.png)
val_batch0_labels.jpg, epochs=5
![1681959040107_3238BEB2-0503-4979-909E-6B78A7322AFE](https://user-images.githubusercontent.com/103250675/233246942-e7f8221a-3e88-4523-9bfa-2bb8a585b0f7.png)
val_batch0_labels.jpg, epochs=10
![1681959067123_9F914F72-C2A2-446f-95E1-0ADCADB14F0C](https://user-images.githubusercontent.com/103250675/233246992-6d770fa1-0b20-44e3-998e-6cf2fd590ace.png)
val_batch0_labels.jpg, epochs=30
![1681959226421_4C065266-12FD-4b58-9DE6-74E62A52B41E](https://user-images.githubusercontent.com/103250675/233247039-1031ca72-4bc0-4a1e-bcc3-dfe37d22cf76.png)
As the batch increases，The key points in the tag are brought closer and closer together in the image.This is also the case in my training. The test box can be fitted, but the point loss will not decrease after about 10 batches of key points, resulting in a complete failure to fit the points. I didn't make any changes to the training parameters。

**Comment by Laughing-q** [comment]
@Yangninan Hey, I'd like to know what's your package version? this probably is related to a bug in our previous version. And it had been fixed in the latest version.

**Comment by Yangninan** [comment]
Thanks for your answer, this problem has been solved in the latest code.

**Comment by jazzseow** [comment]
Yes, I tried it on ultralytics==8.0.82, and it works fine now. Thank you

**Comment by glenn-jocher** [comment]
@jazzseow glad to hear that the issue got resolved with the latest code version. It is common to encounter bugs while using a particular package and keeping the software up-to-date is always helpful to ensure that you are benefiting from the latest improvements and bug fixes. As always, if you have any further concerns or issues, feel free to ask for our help.
